src/setuptools_pyproject_migration/__init__.py

```diff
@@ -91,8 +91,67 @@ def _generate(self) -> Pyproject:
         if dependencies:
             pyproject["project"]["dependencies"] = dependencies
 
+        entrypoints = self._generate_entrypoints(dist.entry_points)
+
+        # GUI scripts and console scripts go separate in dedicated locations.
+        if "console_scripts" in entrypoints:
+            pyproject["project"]["scripts"] = entrypoints.pop("console_scripts")
+
+        if "gui_scripts" in entrypoints:
+            pyproject["project"]["gui-scripts"] = entrypoints.pop("gui_scripts")
+
+        # Anything left over gets put in entry-points
+        if entrypoints:
+            pyproject["project"]["entry-points"] = entrypoints
+
         return pyproject
 
+    @staticmethod
+    def _parse_entrypoint(entrypoint: str) -> Tuple[str, str]:
+        """
+        Extract the entry point and name from the string.
+
+        .. code-block:: python
+
+            WritePyproject._parse_entrypoint("hello-world = timmins:hello_world")
+            # returns ('hello-world', 'timmins:hello_world')
+
+            WritePyproject._parse_entrypoint("hello-world")
+            # throws ValueError: Entry point 'hello-world' is not of the form 'name = target'
```

Comment:
I would still really recommend making these doctests, unless you have some reason not to. I think we gain quite a bit by making them real runnable tests, e.g. if the behavior of this function ever changes in a way that causes the code to become inaccurate, then if they are doctests it'll be caught by automatic testing, whereas with this syntax we rely on someone trawling through the code and noticing.

```suggestion
        >>> WritePyproject._parse_entrypoint("hello-world = timmins:hello_world")
        {'hello-world': 'timmins:hello_world'}
        >>> WritePyproject._parse_entrypoint("hello-world")
        Traceback (most recent call last):
            ...
        ValueError: Entry point 'hello-world' is not of the form 'name = target'
```

...unless this syntax actually _is_ recognized by doctest? (If so I just haven't seen it before)

Reply:
Yeah, I guess I was trying to think how the Sphinx docs engine would render it or how to mark it up to indicate to the documentation system "this is example code", and `>>>` isn't valid Python syntax.

No biggie though, we'll give the "raw" formatting method a try and see how it goes.

Reply:
Hmm, good point, I totally forgot to consider how Sphinx renders this. Whoops :sweat_smile: But there must be a way to keep the `>>>` prefix (which AFAIK doctest needs to recognize it as an executable test) while also ensuring that Sphinx will produce something reasonable. I mean, I'm pretty sure I've seen other projects do both. If it doesn't just work as-is, off the top of my head I'm guessing there might be a language like "ipython" or "python prompt" or something that you can use for the code block, which will allow `>>>`. (That's why doctest uses `>>>`, because it's supposed to look like an interactive Python session.)

Anyway, I'm fine either way: we can go ahead with "raw" formatting and see what Sphinx does, and if it doesn't come out looking good we'll fix it up in another PR; or if you prefer, feel free to figure out how to make them both work together, and we'll do that. We can fix up any doctests that have already been merged in another PR.

Reply:
...actually Sphinx doesn't even render documentation for private methods (starting with `_`), it seems. I'm sure there's a configuration setting for that but to me it makes sense to keep it as-is.

I still kind of like _having_ the documentation even if it's not actually rendered. It's handy for people reading the code.